src/test/java/nus/climods/model/module/ModuleTest.java

```diff
@@ -0,0 +1,80 @@
+package nus.climods.model.module;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.openapitools.client.ApiException;
+import org.openapitools.client.api.ModulesApi;
+import org.openapitools.client.model.ModuleInformation;
+import org.openapitools.client.model.SemestersEnum;
+
+class ModuleTest {
+
```

Comment:
not sure if the negative test cases are required, if there is a need for those we can add them during our v1.4

Reply:
yeap good catch i will add it as a TODO